src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.EventDispatch.cs

```diff
@@ -74,6 +74,11 @@ private Task ReturnErrorResponse(string detailedMessage)
             : Task.CompletedTask;
     }
 
+    private void SetNotFoundResponse(object? sender, EventArgs args)
+    {
+        _httpContext.Response.StatusCode = StatusCodes.Status404NotFound;
```

Comment:
There's something "obvious" that I missed here. 

At this point we want to also "signal" the renderer to stop processing any new renders and start the "teardown process".

Reply:
You mean invoking `Renderer.InvokeRenderCompletedCalls`?

Reply:
Nope, that's something that only happens in interactive scenarios.